src/overmap.cpp

```diff
@@ -3981,33 +3833,99 @@ void overmap::good_river(int x, int y, int z)
     (0,2)(1,2)(2,2)    (-2,0)(-2,1)(-2,2)    (-2,2)(-1,2)(0,2)
 */
 
-inline tripoint rotate_tripoint(tripoint p, int rotations)
+point om_direction::rotate( const point &p, type dir )
 {
-    if(rotations == 1) {
-        return tripoint(-1 * p.y, p.x, p.z);
-    } else if(rotations == 2) {
-        return tripoint(-1 * p.x, -1 * p.y, p.z);
-    } else if(rotations == 3) {
-        return tripoint(p.y, -1 * p.x, p.z);
+    switch( dir ) {
+        case type::invalid:
+            debugmsg( "Invalid overmap rotation (%d).", dir );
+            // Intentional fallthrough.
+        case type::north:
+            return p;
+        case type::east:
+            return point( -p.y, p.x );
+        case type::south:
+            return point( -p.x, -p.y );
+        case type::west:
+            return point( p.y, -p.x );
     }
```

Comment:
> there is no check when casting int to enum on most compilers.

Yes, but Implicit casts from `int` are prohibited by `enum class` (I don't know if the compiler takes that into account). For now, there are only two places in the code with explicit casts, but they're well-encapsulated and both have the [manual] check. Nonetheless, I'm fixing the `switch`: it's a right thing to do.